INSTALL

```diff
@@ -1,118 +1,324 @@
-In this case Airflow will be installed in non-editable mode with all providers installed from the sources.
-Additionally `provider.yaml` files will also be copied to providers folders which will make the providers
-discoverable by Airflow even if they are not installed from packages in this case.
-
-INSTALL_PROVIDERS_FROM_SOURCES="true" pip install . \
-  --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-3.8.txt"
-
-Airflow can be installed with extras to install some additional features (for example 'async' or 'doc' or
-to install automatically providers and all dependencies needed by that provider:
-
-pip install .[async,google,amazon] \
-  --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-3.8.txt"
-
-The list of available extras:
-
-# START EXTRAS HERE
-aiobotocore, airbyte, alibaba, all, all_dbs, amazon, apache.atlas, apache.beam, apache.cassandra,
-apache.drill, apache.druid, apache.flink, apache.hdfs, apache.hive, apache.impala, apache.kafka,
-apache.kylin, apache.livy, apache.pig, apache.pinot, apache.spark, apache.webhdfs, apprise,
-arangodb, asana, async, atlas, atlassian.jira, aws, azure, cassandra, celery, cgroups, cloudant,
-cncf.kubernetes, cohere, common.io, common.sql, crypto, databricks, datadog, dbt.cloud,
-deprecated_api, devel, devel_all, devel_ci, devel_hadoop, dingding, discord, doc, doc_gen, docker,
-druid, elasticsearch, exasol, fab, facebook, ftp, gcp, gcp_api, github, github_enterprise, google,
-google_auth, grpc, hashicorp, hdfs, hive, http, imap, influxdb, jdbc, jenkins, kerberos, kubernetes,
-ldap, leveldb, microsoft.azure, microsoft.mssql, microsoft.psrp, microsoft.winrm, mongo, mssql,
-mysql, neo4j, odbc, openai, openfaas, openlineage, opensearch, opsgenie, oracle, otel, pagerduty,
-pandas, papermill, password, pgvector, pinecone, pinot, postgres, presto, rabbitmq, redis, s3, s3fs,
-salesforce, samba, saml, segment, sendgrid, sentry, sftp, singularity, slack, smtp, snowflake,
-spark, sqlite, ssh, statsd, tableau, tabular, telegram, trino, vertica, virtualenv, weaviate,
-webhdfs, winrm, yandex, zendesk
-# END EXTRAS HERE
-
-# For installing Airflow in development environments - see CONTRIBUTING.rst
-
-# COMPILING FRONT-END ASSETS (in case you see "Please make sure to build the frontend in static/ directory and then restart the server")
-# Optional : Installing yarn - https://classic.yarnpkg.com/en/docs/install
-
-python setup.py compile_assets
+  --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-main/constraints-no-providers-3.8.txt"
+
+## All airflow extras
+
+Airflow has a number of extras that you can install to get additional dependencies. They sometimes install
+providers, sometimes enable other features where packages are not installed by default.
+
+You can read more about those extras in the extras reference:
+https://airflow.apache.org/docs/apache-airflow/stable/extra-packages-ref.html
```

Comment:
Because INSTALL is meant to be embedded in `source.tar.gz` package distributed via `dowloads.apache.org` - targetted for this installation option: https://airflow.apache.org/docs/apache-airflow/stable/installation/index.html#using-released-sources

Whie `docs` packages are part of this package, You would really have to build the docs and host them locally to be able to view the docs in user-friendly, so for the convenience, we mention the actual docs where ASF hosts documentation.

More context for the interested ones:

The latest packages can be found here: https://downloads.apache.org/airflow/2.8.0/

This is a rarely used option by our users - most of them install airflow via `PyPI`, using .whl packages. 

But there are a number of  users who wants to do it. For example there are are upstream package distributors - various linux distributions, conda, brew, etc. etc. etc. that are preparing their own packages from sources and host them in their repo. Also there are some users who are focused on provenance or simply want to build packages from pure sources + "official" tools. We had quite a few such users who communic

And for the Apache Software Foundation - source packages which are signed, checksummed and released via dowloads.apache.org are the only "official" packages and the only "legally binding" packages that are released by the Foundation. All the other packages are "convenience" binaries and you should be able to build such convenience packages (for example .sdist, .whl, our docker reference images) only using sources released via `downloads` - so the source.tar.gz package should be "enough" to rebuild Airflow (having some common and renowned build tools available). So this is important to have INSTALL documentation that describes how to build Airflow, however since we are referring official Airflow documentation - it's easier to just let the user know where to look in the "official" docs rather than ask the user to look at the documentaiton that needs to be built and hosted locally.